[DDW-665] Make the Wallets Dropdown agnostic

daniloprates opened the pull request:
This PR makes the Wallets Dropdown agnostic.

At the moment, we are using the `WalletsDropdown` component for both Wallets and Assets, adding an unwanted huge amount of code. This PR creates an agnostic component called `ItemsDropdown` which receives a normal `options` list as prop, with an extra item called `details`. The `label` will be the top content and the `detail`, if exists, will be the bottom content.

On top of that, `WalletsDropdown` and `AssetsDropdown` extend `ItemsDropdown` component and do a simple job of receiving a list of wallets or assets and translating them into the label/detail format. All three components are much smaller than the current existing one.

One last thing these new components have is a default search function for each, which means adding a simple `hasSearch` to the `AssetsDropdown` will automatically search for in asset property.

## Testing

There wasn't any functionality change, therefore for testing we only need to make sure the existing instances work correctly (some screens, like voting, can only be tested on Storybook). The only new behavior is the search bar for the Assets Token, in the Send screen.

### Storybook

* [Generic Items Dropdown](https://deploy-preview-2540--daedalus-storybook.netlify.app/?path=/story/common-itemsdropdown--generic#themeName=Cardano&localeName=English&osName=Windows)
* [Wallets Dropdown](https://deploy-preview-2540--daedalus-storybook.netlify.app/?path=/story/common-itemsdropdown--wallets#themeName=Cardano&localeName=English&osName=Windows)
* [Assets Dropdown](https://deploy-preview-2540--daedalus-storybook.netlify.app/?path=/story/common-itemsdropdown--assets#themeName=Cardano&localeName=English&osName=Windows)




## Todos

- [x] Create an agnostic `ItemDropdown` component
- [x] Implement `WalletsDropdown` using the `ItemDropdown` component
- [x] Implement `AssetsDropdown` using the `ItemDropdown` component

## Screenshots

![Screen Shot 2021-05-03 at 16 03 17](https://user-images.githubusercontent.com/1504716/116921640-a1515d00-ac2a-11eb-9ace-1d91743f4cab.png)
![Screen Shot 2021-05-03 at 16 04 41](https://user-images.githubusercontent.com/1504716/116921645-a1e9f380-ac2a-11eb-8c1e-1329a87e1e19.png)
![Screen Shot 2021-05-03 at 16 08 00](https://user-images.githubusercontent.com/1504716/116921649-a2828a00-ac2a-11eb-9c13-e4e0c891338a.png)
![Screen Shot 2021-05-03 at 16 08 10](https://user-images.githubusercontent.com/1504716/116921650-a2828a00-ac2a-11eb-94b7-5df2e338eddd.png)
![Screen Shot 2021-05-03 at 16 08 31](https://user-images.githubusercontent.com/1504716/116921651-a31b2080-ac2a-11eb-9e5d-2a8cf191b450.png)
![Screen Shot 2021-05-03 at 16 08 39](https://user-images.githubusercontent.com/1504716/116921654-a31b2080-ac2a-11eb-9e76-ac9aecfcdb91.png)
![Screen Shot 2021-05-03 at 16 12 22](https://user-images.githubusercontent.com/1504716/116921657-a3b3b700-ac2a-11eb-9ed2-384b0cb02cd3.png)
![Screen Shot 2021-05-03 at 16 12 33](https://user-images.githubusercontent.com/1504716/116921660-a3b3b700-ac2a-11eb-89ea-08fb36e25f09.png)
![Screen Shot 2021-05-03 at 15 58 28](https://user-images.githubusercontent.com/1504716/116921630-9d253f80-ac2a-11eb-8344-ab80dc278b36.png)
![Screen Shot 2021-05-03 at 16 00 31](https://user-images.githubusercontent.com/1504716/116921636-9f879980-ac2a-11eb-9c9e-e402ed5809bd.png)
![Screen Shot 2021-05-03 at 16 20 36](https://user-images.githubusercontent.com/1504716/116922316-921edf00-ac2b-11eb-9bb2-fde117148eb6.png)
![Screen Shot 2021-05-03 at 16 20 49](https://user-images.githubusercontent.com/1504716/116922319-94813900-ac2b-11eb-9b24-b50a1e23ff4a.png)


---
## Test Scenarios

**Scenario 1 - Validate dropdown behavior and layout**
```gherkin
Given that I am on |Screen|
When I check the |Dropdown|
Then I can see the behavior and layout are as observed in the last version on production

```
Screen | Dropdown
-- | --
Initial settings | Language
Initial settings | Number Format
Initial settings | Date Format
Initial settings | Time Format
Settings/General | Language
Settings/General | Number Format
Settings/General | Date Format
Settings/General | Time Format
Settings/Wallets | Select Currency
Settings/Stake Pools | Off-chain metadata server
Delegation Center/Delegation Wizard | Wallet
Stake Pools | Wallets
ITN Rewards | Redeem rewards to
Wallet/Filter transactions | Select Time Range
Wallet/Send | Native Token
Transfer Funds | Shelley Compatible Wallet
Native token settings | Number of decimal places


---
## Issues Summary

**New**

- Spinner is not aligned with the text. Check [this](https://files.slack.com/files-pri/T0N639Z4N-F02228VLWDP/aligned.gif) video.

**Fixed** 
- The "syncing" on the tooltip should start with uppercase. Check [this screenshot](https://drive.google.com/file/d/1Xfa03pmxJpH2N_Gt0-Qt01fcf14sCUvi/view?usp=sharing). Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-837181976).

- Also, for similar tooltips on other features, the syncing percentage is also displayed. Check [this screenshot](https://drive.google.com/file/d/1OxB_ef6P6lWUTSTm_IpxQy5B0j_QnF77/view?usp=sharing). Could that be implemented on the other tooltips too so it's more consistent? Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-837181976).

- The "Select wallet" dropdown is enabled on ITN rewards when the recovery phrase field is empty(It was disabled before, until a valid recovery phrase was inserted). [Check this screenshot](https://drive.google.com/file/d/1Cfkzej26eR2uW2xoAQS_9kUjvmczPpLS/view?usp=sharing). Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-837181976).

- Remove the word "Syncing" from the list and add it to a tooltip on the spinner. Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-836558689).

- Remove the SP ticker for the selected wallet from the dropdown on the "Stake pools" scren. Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-836558689).

- Balance of the token on the "Send" screen is only visible on the list, not in the preview. Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-836558689).

- Under "Stake Pools" tab, the layout of the drop down is not quite right. Please see [here](https://drive.google.com/file/d/1WkpL3ndQWYSjOypU5EpkLqHZT0KksHWq/view?usp=sharing). Also there's an extra space between wallet's name and the down arrow, please see [here](https://drive.google.com/file/d/1IA575VcsTBb2QvaQMEATE3vzx74KxylZ/view?usp=sharing). Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-834546510).

- In storybook those ( [Generic Items Dropdown](https://deploy-preview-2540--daedalus-storybook.netlify.app/?path=/story/common-itemsdropdown--generic#themeName=Cardano&localeName=English&osName=Windows), [Wallets Dropdown](https://deploy-preview-2540--daedalus-storybook.netlify.app/?path=/story/common-itemsdropdown--wallets#themeName=Cardano&localeName=English&osName=Windows) and [Assets Dropdown](https://deploy-preview-2540--daedalus-storybook.netlify.app/?path=/story/common-itemsdropdown--assets#themeName=Cardano&localeName=English&osName=Windows) ) dropdowns are big. Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-831738687).

- When the user uses the dropdown for wallets in Redeem ITN Rewards and the dropdown for wallets in Delegation Centre the spinner for loading wallets is not displaying like it used to.[Video](https://drive.google.com/file/d/1YzUQhwmWb1xPDYlgDam7a5TGsRcKQAea/view?usp=sharing). Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-831809170).

- Wallet summary in storybook seems to be broken. Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#pullrequestreview-651448442).

- I can still not see the spinner for the dropdown in Delegation center [Video](https://drive.google.com/file/d/1aq-FsVwVgcslpssNFJPXPMLiRiI2SwdV/view?usp=sharing). The spinner in Itn Rewards Redemption can be seen but is misaligned [Video](https://drive.google.com/file/d/1mkhkiMsraYTL_s4lKqqmb3qrtF2Pfu5_/view?usp=sharing). Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-832555454).

- On the "Currency" dropdown, with no results visible:

  - wrong "No results" position
  - corner radius for dropdown is bigger that for input (should be the same)
  - letter "g" in input seems to be cropped. Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-832565437).

- On Delegation wizard the syncing spinner shifted too far from right (it should be centered with checkmark). Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-832565437).

- I found this issue on Stake Pools tab whereby the wallets are not populated and the layout is off. Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-832675922).

- Syncing indication has more priority than selection, so we can show only spinner and drop checkmark. Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-833393183).

- The colour of the spinner is too light coloured in Redeem ITN Rewards screen for themes Light blue and Cardano [Video](https://drive.google.com/file/d/1PtAIP7d8pRGp1CmvUVlFXEADFFPC9ocG/view?usp=sharing)  [Video2](https://drive.google.com/file/d/1LZurM3A1QvfYiWy-ZH_m-tGFVMEV0KwR/view?usp=sharing). Could you make it more visible please. Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-833588004).

- The dropdown for native tokens looks bigger than before and it has a border that exceeds the size of the input. Check [this screenshot](https://drive.google.com/file/d/1p7Lq1rbD626_1iAdqsGtJR4xFaEtqXhJ/view?usp=sharing). Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-833818047).

- For loading wallets, the spinner should be aligned with the checkmark. Check [this screenshot](https://drive.google.com/file/d/1FKCUnBcp5AKowI1lA4OFdAr3I7JjuF81/view?usp=sharing). And for [this one](https://drive.google.com/file/d/1NlLtZgpdy-_-umZtjt2gho5kvmWooYLK/view?usp=sharing), there is no check, but the spinner is too far from the right. Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-833818047).

**To be addressed on another task**

- As the Japanese characters are bigger when a wallet has a long name in this language, the arrow and text overlap. Check [this video](https://drive.google.com/file/d/1oo1YVR6HgPIL-HRIlj1v9_gXJZfe3Qm_/view?usp=sharing). Reported [here](https://github.com/input-output-hk/daedalus/pull/2540#issuecomment-834727075).

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

## Review Checklist

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

alexander-rukin commented:
<img width="265" alt="image" src="https://user-images.githubusercontent.com/2539345/116971878-e923c300-acc2-11eb-8a52-89300739379d.png">
@daniloprates i storybook those ( [Generic Items Dropdown](https://deploy-preview-2540--daedalus-storybook.netlify.app/?path=/story/common-itemsdropdown--generic#themeName=Cardano&localeName=English&osName=Windows) [Wallets Dropdown](https://deploy-preview-2540--daedalus-storybook.netlify.app/?path=/story/common-itemsdropdown--wallets#themeName=Cardano&localeName=English&osName=Windows) [Assets Dropdown](https://deploy-preview-2540--daedalus-storybook.netlify.app/?path=/story/common-itemsdropdown--assets#themeName=Cardano&localeName=English&osName=Windows) ) dropdowns are big. Is this a bug?

ManusMcCole commented:
Hi @daniloprates. Here is the testing report. Tested on build 17836

When the user uses the dropdown for wallets in Redeem ITN Rewards and the dropdown for wallets in delegation centre the spinner for loading wallets is not displaying like it used to.[Video](https://drive.google.com/file/d/1YzUQhwmWb1xPDYlgDam7a5TGsRcKQAea/view?usp=sharing)

DominikGuzei commented on source/renderer/app/components/widgets/forms/ItemsDropdown.js:
```diff
@@ -0,0 +1,56 @@
+  options: Array<ItemDropdown>,
+};
+
+export const onSearchItemsDropdown = (
```
Same here … something like `filterOptionsBySearchValue`

daniloprates replied:
I'd keep it for the [same reason](https://github.com/input-output-hk/daedalus/pull/2540#discussion_r626009138)

DominikGuzei commented on storybook/stories/wallets/legacyWallets/TransferFunds.stories.js:
```diff
@@ -44,6 +44,7 @@ storiesOf('Wallets|Legacy Wallets', module)
       ...sourceWalletSelect,
       amount: new BigNumber(sourceWalletSelect.amount),
     };
+    console.log('WALLETS_V2', WALLETS_V2);
```
This can probably be removed 😉

daniloprates replied:
I ❤️  console logs 😅

DominikGuzei reviewed (changes requested):
Great improvement, the code looks much cleaner now 🎉 I have only added some small comments

Wallet summary in storybook seems to be broken (not sure if it's due to your changes)
![grafik](https://user-images.githubusercontent.com/172414/117046884-83aaf300-ad11-11eb-8799-ca0d56263085.png)

daniloprates commented on source/renderer/app/components/widgets/forms/AssetsDropdown.js:
```diff
@@ -0,0 +1,71 @@
+  assets?: Array<$Shape<WalletSummaryAsset>>,
+};
+
+export const onSearchAssetsDropdown = (
```
I'd keep it as it is, because the prop name in the RP's `Select` component is `onSearch`, so `onSearchAssetsDropdown` is more similar. The params position are also derived from there, and the `searchValue` is the first one because in some cases you may simply ignore the array in second one, when you have already the list in a better format.

DominikGuzei replied:
Ok

daniloprates commented:
@alexander-rukin @ManusMcCole fixed the syncing and heights issues

ManusMcCole commented:
Hi @daniloprates.Tested on build 17845

I can still not see the spinner for the dropdown in Delegation centre [Video](https://drive.google.com/file/d/1mkhkiMsraYTL_s4lKqqmb3qrtF2Pfu5_/view?usp=sharing).The spinner in Itn Rewards Redemption can be seen but is is misaligned [Video](https://drive.google.com/file/d/1aq-FsVwVgcslpssNFJPXPMLiRiI2SwdV/view?usp=sharing)

alexander-rukin commented:
<img width="258" alt="image" src="https://user-images.githubusercontent.com/2539345/117124961-a389f780-ada1-11eb-9b92-cc08b166154d.png">

I see 2 issues here: 
- wrong "No results" position 
- corner radius for dropdown is bigger that for input (should be the same)

miorsufianiohk commented:
Hi @daniloprates , on build 17845,  in addition to what the others have found so far. I found this issue on Stake Pools tab whereby the wallets are not populated and the layout is off.

![Screenshot 2021-05-05 at 14 02 50](https://user-images.githubusercontent.com/67595081/117146096-c28d8700-adab-11eb-8125-6bd21a26ef2d.png)
![Screenshot 2021-05-05 at 14 02 35](https://user-images.githubusercontent.com/67595081/117146118-c8836800-adab-11eb-938e-7a181af8751d.png)

daniloprates commented:
@alexander-rukin @ManusMcCole @miorsufianiohk thanks for the great catches! The are all fixed, except for one scenario, that I'm not sure how to handle:

![image](https://user-images.githubusercontent.com/1504716/117202935-d396c200-adc4-11eb-8bbc-23df187f0ab3.png)

Maybe syncing wallets shouldn't be selectable?

alexander-rukin commented:
> @alexander-rukin @ManusMcCole @miorsufianiohk thanks for the great catches! The are all fixed, except for one scenario, that I'm not sure how to handle:
> 
> ![image](https://user-images.githubusercontent.com/1504716/117202935-d396c200-adc4-11eb-8bbc-23df187f0ab3.png)
> 
> Maybe syncing wallets shouldn't be selectable?

syncing indication has more priority than selection, so we can show only spinner and drop checkmark

daniloprates commented:
> > @alexander-rukin @ManusMcCole @miorsufianiohk thanks for the great catches! The are all fixed, except for one scenario, that I'm not sure how to handle:
> > ![image](https://user-images.githubusercontent.com/1504716/117202935-d396c200-adc4-11eb-8bbc-23df187f0ab3.png)
> > Maybe syncing wallets shouldn't be selectable?
> 
> syncing indication has more priority than selection, so we can show only spinner and drop checkmark

Cool, removed the label for this instance of the dropdown

@alexander-rukin @ManusMcCole @miorsufianiohk it's ready to be tested

daniloprates commented:
@DominikGuzei the Wallets Summary story was broken by another PR, but I've [fixed it](https://deploy-preview-2540--daedalus-storybook.netlify.app/?path=/story/wallets-summary--wallet-summary#themeName=Cardano&localeName=English&osName=Windows) :)

ManusMcCole commented:
Hi @daniloprates.Tested on build 17875

The changes look good I just have one recommendation. The colour of the spinner is too light coloured in Redeem ITN Rewards screen for themes Light blue and Cardano [Video](https://drive.google.com/file/d/1PtAIP7d8pRGp1CmvUVlFXEADFFPC9ocG/view?usp=sharing)  [Video2](https://drive.google.com/file/d/1LZurM3A1QvfYiWy-ZH_m-tGFVMEV0KwR/view?usp=sharing). Could you make it more visible please

gabriela-ponce commented:
@daniloprates A few issues on this PR:

- The dropdown for native tokens looks bigger than before and it has a border that exceeds the size of the input. Check [this screenshot](https://drive.google.com/file/d/1p7Lq1rbD626_1iAdqsGtJR4xFaEtqXhJ/view?usp=sharing).
- For loading wallets, the spinner should be aligned with the checkmark. Check [this screenshot](https://drive.google.com/file/d/1FKCUnBcp5AKowI1lA4OFdAr3I7JjuF81/view?usp=sharing). And for [this one](https://drive.google.com/file/d/1NlLtZgpdy-_-umZtjt2gho5kvmWooYLK/view?usp=sharing), there is no check, but the spinner is too far from the right.

daniloprates commented:
> @daniloprates A few issues on this PR:
> 
> * The dropdown for native tokens looks bigger than before and it has a border that exceeds the size of the input. Check [this screenshot](https://drive.google.com/file/d/1p7Lq1rbD626_1iAdqsGtJR4xFaEtqXhJ/view?usp=sharing).
> * For loading wallets, the spinner should be aligned with the checkmark. Check [this screenshot](https://drive.google.com/file/d/1FKCUnBcp5AKowI1lA4OFdAr3I7JjuF81/view?usp=sharing). And for [this one](https://drive.google.com/file/d/1NlLtZgpdy-_-umZtjt2gho5kvmWooYLK/view?usp=sharing), there is no check, but the spinner is too far from the right.

Thanks @gabriela-ponce. I wasn't sure how to deal with spinner + checkmark, but realized in the current version the selected option doesn't have the spinner, so it's how I fixed it. Assets Dropdown also fixed.

miorsufianiohk commented:
Hi @daniloprates , on build 17903, under "Stake Pools" tab, the layout of the drop down is not quite right. Please see [here](https://drive.google.com/file/d/13L_YItK6lNduxfM2mk_8z-6TBdqxzwcv/view?usp=sharing). Also there's an extra space between wallet's name and the down arrow, please see [here](https://drive.google.com/file/d/1WkpL3ndQWYSjOypU5EpkLqHZT0KksHWq/view?usp=sharing).

daniloprates commented:
@thanks @miorsufianiohk. It's fixed.

gabriela-ponce commented:
@daniloprates The latest fixes look great. I just found another small issue with the stake pool dropdown.

- As the Japanese characters are bigger when a wallet has a long name in this language, the arrow and text overlap. Check [this video](https://drive.google.com/file/d/1oo1YVR6HgPIL-HRIlj1v9_gXJZfe3Qm_/view?usp=sharing). However, this also happens also in production. Do you think this can be addressed here or another task is required?

daniloprates commented:
> @daniloprates The latest fixes look great. I just found another small issue with the stake pool dropdown.
> 
> * As the Japanese characters are bigger when a wallet has a long name in this language, the arrow and text overlap. Check [this video](https://drive.google.com/file/d/1oo1YVR6HgPIL-HRIlj1v9_gXJZfe3Qm_/view?usp=sharing). However, this also happens also in production. Do you think this can be addressed here or another task is required?

@gabriela-ponce I had created [a card](https://jira.iohk.io/browse/DDW-677) for overall improvements in this screen, and now added the issue you described. It can be fixed there.

ManusMcCole reviewed (approved):
Good work @daniloprates 👍

miorsufianiohk reviewed (approved):
LGTM. Tested on 17908. 👍

nikolaglumac commented:
@daniloprates I have a few questions and would like you to answer them before merging this PR 🙏 

![1](https://user-images.githubusercontent.com/376611/117649770-37880480-b190-11eb-9636-57c61e4efe9b.png)
![2](https://user-images.githubusercontent.com/376611/117649780-39ea5e80-b190-11eb-8e43-4b1719676b16.png)
![3](https://user-images.githubusercontent.com/376611/117649784-3a82f500-b190-11eb-9ebc-95c4bea28a96.png)
![4](https://user-images.githubusercontent.com/376611/117649786-3b1b8b80-b190-11eb-86f6-b3cff7700c1e.png)
![5](https://user-images.githubusercontent.com/376611/117649789-3bb42200-b190-11eb-88b2-5cc053ae39db.png)

cc @alexander-rukin

daniloprates commented:
> @daniloprates I have a few questions and would like you to answer them before merging this PR 🙏
> 

@nikolaglumac 

**1** - No, but I don't see why it has in some cases and not in others. Shouldn't we add/remove it from all instances?

_Have label_
* staking/delegation-setup-wizard/DelegationStepsChooseWalletDialog.js
* staking/redeem-itn-rewards/Step1ConfigurationDialog.js
* wallet/send-form/AssetInput.js

_Don't have label_
* staking/stake-pools/StakePoolsRanking.js
* voting/voting-registration-wizard-steps/VotingRegistrationStepsChooseWallet.js
* wallet/transfer-funds/TransferFundsStep1Dialog.js

**2** and **4** - Fixed

**3** - It's broken in Develop and I'd rather not mix with this PR

gabriela-ponce commented:
@daniloprates a few minor issues and suggestions regarding the latest changes:

- The "syncing" on the tooltip should start with uppercase. Check [this screenshot](https://drive.google.com/file/d/1Xfa03pmxJpH2N_Gt0-Qt01fcf14sCUvi/view?usp=sharing).

- Also, for similar tooltips on other features, the syncing percentage is also displayed. Check [this screenshot](https://drive.google.com/file/d/1OxB_ef6P6lWUTSTm_IpxQy5B0j_QnF77/view?usp=sharing). Could that be implemented on the other tooltips too so it's more consistent? 

- The "Select wallet" dropdown is enabled on ITN rewards when the recovery phrase field is empty(It was disabled before, until a valid recovery phrase was inserted). [Check this screenshot](https://drive.google.com/file/d/1Cfkzej26eR2uW2xoAQS_9kUjvmczPpLS/view?usp=sharing).

gabriela-ponce commented:
@daniloprates The small spinner looks good! Just a few small issues:

- The spinner for the light blue theme on "ITN rewards" is hard to see, especially on hover. Check [this screenshot.](https://drive.google.com/file/d/19obbB2iqlg66xaadp6GujLSL4HqLNrTz/view?usp=sharing)

- The spinner is not properly aligned with the text. Check [this screenshot](https://drive.google.com/file/d/1sc8MKyS8LWrJuHOncIJFTiEVUXOhxGRf/view?usp=sharing).

Also this issue still reproduces on the latest build:

- The "Select wallet" dropdown is enabled on ITN rewards when the recovery phrase field is empty(It was disabled before, until a valid recovery phrase was inserted). [Check this screenshot](https://drive.google.com/file/d/1Cfkzej26eR2uW2xoAQS_9kUjvmczPpLS/view?usp=sharing).

daniloprates commented:
> * Check this screenshot

Thanks @gabriela-ponce, just pushed a fix for the 3 above.

nikolaglumac commented:
@gabriela-ponce is this one OK now?

gabriela-ponce commented:
@nikolaglumac I'm still checking this PR, would update the status later 👍

gabriela-ponce commented:
@daniloprates The spinner on [ITN rewards](https://drive.google.com/file/d/1Rx6WnKHKwqphqI0YZGMlc8zXB23xA0PZ/view?usp=sharing) looks ok now but on [delegation center](https://drive.google.com/file/d/13dMK5tidGMmvarYjPFG3NdL4Oy9lP_ic/view?usp=sharing) is too high in comparison.

daniloprates commented:
@gabriela-ponce just pushed a fix for the spinner:

### Delegation

**Wrong**
![B-wizard](https://user-images.githubusercontent.com/1504716/118290362-be005700-b4ac-11eb-8b98-26213b72969d.png)

**Fixed**
![A-wizard](https://user-images.githubusercontent.com/1504716/118290357-bd67c080-b4ac-11eb-8f2c-cbd6cfe2b554.png)

### ITN

**Before - already correct**
![B-itn](https://user-images.githubusercontent.com/1504716/118290359-be005700-b4ac-11eb-8070-3e0aace1a5e1.png)

**After latest changes, remain correct**
![A-itn](https://user-images.githubusercontent.com/1504716/118290350-bb9dfd00-b4ac-11eb-8e39-2eaea1127267.png)

nikolaglumac reviewed (approved):
Amazing work @daniloprates! This was long overdue 😅

gabriela-ponce reviewed (approved):
Looks great
